File: src/main/java/org/example/bankramenserver/infrastructure/mcp/auth/AgentConnectionRepository.java
package org.example.bankramenserver.infrastructure.mcp.auth;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface AgentConnectionRepository extends JpaRepository<AgentConnection, String> {

    Optional<AgentConnection> findByTokenHash(String tokenHash);
}

File: src/main/kotlin/org/example/balogserver/infrastructure/mcp/auth/AgentConnectionRepository.kt
package org.example.balogserver.infrastructure.mcp.auth

import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface AgentConnectionRepository : JpaRepository<AgentConnection, String> {
    fun findByTokenHash(tokenHash: String?): Optional<AgentConnection>
}
